src/main/java/org/datadog/jmxfetch/util/CustomLogger.java

```diff
@@ -1,123 +1,166 @@
+            private static final String format = JDK14_LAYOUT;
+
+            private String simpleClassName(String str) {
+                int start = str.lastIndexOf('.');
+                int end = str.indexOf('$');
+                if (start == -1 || start + 1 == str.length()) {
+                    return str;
+                }
+                if (end == -1 || end <= start || end > str.length()) {
+                    end = str.length();
+                }
+                return str.substring(start + 1, end);
+            }
 
-        String logPattern = logFormatRfc3339 ? LAYOUT_RFC3339 : LAYOUT;
-
-        PatternLayout layout = PatternLayout.newBuilder()
-            .withConfiguration(config)
-            .withPattern(logPattern)
-            .build();
-
-        if (logLocation != null
-                && !ConsoleAppender.Target.SYSTEM_ERR.toString().equals(logLocation)
-                && !SYSTEM_ERR_ALT.equals(logLocation)
-                && !ConsoleAppender.Target.SYSTEM_OUT.toString().equals(logLocation)
-                && !SYSTEM_OUT_ALT.equals(logLocation)) {
-
-            target = "FileLogger";
-
-            RollingFileAppender fa = RollingFileAppender.newBuilder()
-                .setConfiguration(config)
-                .withName(target)
-                .withLayout(layout)
-                .withFileName(logLocation)
-                .withFilePattern(logLocation + ".%d")
-                .withPolicy(SizeBasedTriggeringPolicy.createPolicy("5MB"))
-                .withStrategy(DefaultRolloverStrategy.newBuilder().withMax("1").build())
-                .build();
-
-            fa.start();
-            config.addAppender(fa);
-            ctx.getRootLogger().addAppender(config.getAppender(fa.getName()));
-
-            log.info("File Handler set");
-        } else {
-
-            if (logLocation != null
-                    && (ConsoleAppender.Target.SYSTEM_ERR.toString().equals(logLocation)
-                        || SYSTEM_ERR_ALT.equals(logLocation))) {
-
-                ConsoleAppender console = (ConsoleAppender)config.getAppender("CONSOLE");
-                console.stop();
-                config.getRootLogger().removeAppender("CONSOLE");
-                ctx.updateLoggers();
-
-                ConsoleAppender ca = ConsoleAppender.newBuilder()
-                    .setConfiguration(config)
-                    .withName(logLocation)
-                    .setTarget(ConsoleAppender.Target.SYSTEM_ERR)
-                    .withLayout(layout)
-                    .build();
-
-                ca.start();
-                config.addAppender(ca);
-                ctx.getRootLogger().addAppender(config.getAppender(ca.getName()));
+            @Override
+            public synchronized String format(LogRecord lr) {
+                SimpleDateFormat simpleDateFormat = new SimpleDateFormat(dateFormat,
+                                                        Locale.getDefault());
```

Comment:
Can this variable be pulled out of the `@Override` itself since it's in the hot path?

Reply:
I think you're right